demo/webcam_demo_spatiotemporal_det.py

```diff
@@ -0,0 +1,803 @@
+        `display_bboxes`, `stdet_bboxes` and `action_preds`, update `frames`.
+    3) Display Thread: Get data from display queue, show/write frames and
+        delete task.
+    """
+
+    def __init__(self):
+        self.id = -1
+
+        # raw frames, used as human detector input, draw predictions input
+        # and output, display input
+        self.frames = None
+
+        # stdet params
+        self.processed_frames = None  # model inputs
+        self.frames_inds = None  # select frames from processed frames
+        self.img_shape = None  # model inputs, processed frame shape
+        # `action_preds` is `list[list[tuple]]`. The outter brackets indicate
+        # different bboxes and the intter brackets indicate different action
+        # results for the same bbox. tuple contains `class_name` and `score`.
+        self.action_preds = None  # stdet results
+
+        # human detector results
+        self.display_bboxes = None  # bboxes coords for self.frames
+        self.stdet_bboxes = None  # bboxes coords for self.processed_frames
+        self.ratio = None  # processed_frames.shape[:2] / frames.shape[:2]
+
+        # number of overlapping frames with the previous task
+        self.num_buffer_frames = 0
+
+        # for each clip, draw predictions around keyframe with at most
+        # clip_vis_length frames
+        self.clip_vis_length = -1
+
+    def add_frames(self, idx, frames, processed_frames):
+        """Add the clip and corresponding id.
+
+        Args:
+            idx (int): the current index of the clip.
+            frames (list[ndarray]): list of images in "BGR" format.
+            processed_frames (list[ndarray]): list of resize and normed images
+                in "BGR" format.
+        """
+        self.frames = frames
+        self.processed_frames = processed_frames
+        self.id = idx
+        self.img_shape = processed_frames[0].shape[:2]
+
+    def add_bboxes(self, display_bboxes):
+        """Add correspondding bounding boxes."""
+        self.display_bboxes = display_bboxes
+        self.stdet_bboxes = display_bboxes.clone()
+        self.stdet_bboxes[:, ::2] = self.stdet_bboxes[:, ::2] * self.ratio[0]
+        self.stdet_bboxes[:, 1::2] = self.stdet_bboxes[:, 1::2] * self.ratio[1]
+
+    def add_action_preds(self, preds):
+        """Add the corresponding action predictions."""
+        self.action_preds = preds
+
+    def get_model_inputs(self, device):
+        """Convert preprocessed images to MMAction2 STDet model inputs."""
+        cur_frames = [self.processed_frames[idx] for idx in self.frames_inds]
+        input_array = np.stack(cur_frames).transpose((3, 0, 1, 2))[np.newaxis]
+        input_tensor = torch.from_numpy(input_array).to(device)
+        return dict(
+            return_loss=False,
+            img=[input_tensor],
+            proposals=[[self.stdet_bboxes]],
+            img_metas=[[dict(img_shape=self.img_shape)]])
+
+
+class BaseHumanDetector:
```

Comment:
```python
from abc import ABCMeta, abstractmethod
...
class BaseHumanDetector(nn.Module, metaclass=ABCMeta):
```

Reply:
Seems it doesn't need to be an nn.Module.